lib/LANraragi/Utils/Tags.pm

```diff
@@ -87,21 +99,42 @@ sub tags_rules_to_array {
     return @rules;
 }
 
+# build hash_replace rules and return the remaining rules
+sub build_tag_replace_hash {
```

Comment:
Shouldn't this method just be merged with `tags_rules_to_array`, which should then return both the array of standard rules and the hash?  

I don't really see the point in temporarily putting `hash_replace` items in the array just to remove them in the very next function.

Reply:
I am doing this just in case someone, somewhere, sometime, wants to get all tags array😀
you're right, I'll just merge it.

Reply:
oh! I remember why I didn't merge it at that time. The original rule array is needed for persistence [here](https://github.com/Difegue/LANraragi/blob/96a9c33b6625584ccaace95e4131127407fe244f/lib/LANraragi/Controller/Config.pm#L153)

Reply:
Right, I forgot about that 😩 
Well it can't be helped in that case I guess; I don't think it's worth spinning up a secondary database field just to save the hashreplace rules.